api/v1beta2/network_types.go

```diff
@@ -307,8 +308,18 @@ func (v *VPCSpec) IsIPv6Enabled() bool {
 // SubnetSpec configures an AWS Subnet.
 type SubnetSpec struct {
 	// ID defines a unique identifier to reference this resource.
+	// If you're bringing your subnet, set the AWS subnet-id here, it must start with `subnet-`.
```

Comment:
This is an AWS-ism right? We aren't requiring users to go name their subnets in a specific way?

I would expect, when populating the subnets, that I should be able to provide any name and, if the subnet exists and isn't tagged with the CAPI resource management tag (`sigs.k8s.io/cluster-api-provider-aws/cluster`), is left alone, but, if it doesn't exist, is created and tagged, or is that not the expected flow? That was my understanding of how other BYO works

Reply:
I tested this out and understand now, the name that you enter here will become the name of the resource once it's created, but, the id field will then be repopulated by the subnet ID from AWS.

So above questions don't matter, but, how will this work with GitOps? I wouldn't expect the ID field to be overwritten as this might cause issues for someone trying to create their AWSCluster resource via some sort of GitOps flow

What's the motivation for overwriting the ID?

Reply:
Just to add some context. When we made the initial change to lead to this whole feature regression our goal was to make it compatible with ClusterClass. But that basically only meant that CAPA should be compatible with SSA in a way that if an AWSCluster is continuously written by some tool (GitOps, ClusterClass controller or another controller) this tool and the CAPA controller should not continuously fight over field values because the AWSCluster CRD is not compatible with SSA in a way that allows to converge to a final state of the AWSCluster.

Reply:
I think the godoc is outdated a bit. To make this all work we cannot overwrite the ID

Reply:
Some roughly related context: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4474#issuecomment-1702799841